run.py

```diff
@@ -102,54 +98,20 @@ def config_validator():
     parser = argparse.ArgumentParser()
     parser.add_argument('-t', '--testmode', action='store_true',
                         help='Run server in "test mode".')
-    parser.add_argument('--backup', action='store_true',
-                        help='backup existing issues.db.')
     args = parser.parse_args()
 
-    if not args.testmode:
-        # testing the config file.
-        # this file is only important in non-test mode.
-        # in test mode everything must be mocked, 
-        # so there is no external api communication.
-        config_validator()
-
-    if args.testmode:
-        # disable HttpOnly setting for session cookies so Selenium
-        # can interact with them. *ONLY* do this for testing.
-        app.config['SESSION_COOKIE_HTTPONLY'] = False
-        app.config['TESTING'] = True
-        print("Starting server in ~*TEST MODE*~")
-        app.run(host='localhost')
-    elif args.backup:
-        # Check 'BACKUP_DEFAULT_DEST' before backup
-        if 'BACKUP_DEFAULT_DEST' not in app.config.keys():
-            sys.exit('Please define BAKCUP_DEFAULT_DEST in secrets.py.')
-
-        issue_engine = create_engine(
-            'sqlite:///' + os.path.join(app.config['BASE_DIR'], 'issues.db'))
-        issue_session_maker = sessionmaker(autocommit=False,
-                                           autoflush=False,
-                                           bind=issue_engine)
-        issue_db = scoped_session(issue_session_maker)
-        # Take a backup if issues.db has data dump.
-        if (issue_db().execute('select count(*) from webcompat_issues')
-                      .fetchall()[0][0] > 0):
-            if not os.path.exists(app.config['BACKUP_DEFAULT_DEST']):
-                print('Creating backup directory at {}').format(
-                    os.path.join(app.config['BASE_DIR'], 'issues.db'))
-                os.makedirs(app.config['BACKUP_DEFAULT_DEST'])
-            time_stamp = time.strftime('%Y-%m-%dT%H-%M-%S', time.localtime())
-            issue_backup_db = 'issues_' + time_stamp + '.db'
-            os.rename(os.getcwd() + '/issues.db',
-                      app.config['BACKUP_DEFAULT_DEST'] + issue_backup_db)
-            # Retain last 3 recent backup files
-            backup_files = os.listdir(app.config['BACKUP_DEFAULT_DEST'])
-            backup_files.sort()
-            for old_file in backup_files[:-3]:
-                os.remove(app.config['BACKUP_DEFAULT_DEST'] + old_file)
+    if check_pip_deps():
+        if not args.testmode:
+            # testing the config file.
+            # this file is only important in non-test mode.
+            # in test mode everything must be mocked,
+            # so there is no external api communication.
+            config_validator()
+            app.run(host='localhost')
```

Comment:
(this isn't related to this review... but I was trying to remember why we made this change. I can't recall... it was something related to tests failing maybe? Like, if you used 127.0.0.1 instead of localhost? All I found in history was the commit message @ https://github.com/webcompat/webcompat.com/commit/feddc52c66c60a9f6b1baf164e495405613caa34....)